### DISCUSSION TITLE
Difference between MetricReduction and LossReduction

### DISCUSSION THREAD

**Discussion by anais2390** [Project-MONAI/MONAI#3640] [Q&A]
Hi everyone !

I am currently working on a 2D binary segmentation problem and I am using the dice coefficient to evaluate performance during training. To do so I use `MeanDice()`  combined with a `SupervisedEvaluator`:

`val_metric = {'mean_dice':MeanDice()}`
` evaluator= create_supervised_evaluator(net, val_metric , device, non_blocking=True,
                                            output_transform=lambda x, y, y_pred: ([postproc_pred(i) for i in decollate_batch(y_pred)], [postproc_label(i) for i in decollate_batch(y)]),
                                            prepare_batch=prepare_batch)`

As a double check I also add the `DiceLoss ` class to my evaluator such as : 

`val_metric = {'diceloss':1 - Loss(DiceLoss())}`

However, the results return by the two metrics are highly different, for exemple : 
`INFO:trainer:Epoch[1]: mean_dice: 0.492 diceloss: 0.176 `

I suppose this difference comes from the Reduction step but i can't figure out why ? 
Could you bring me some highlights ? 

Thanks !

**Comment by Nic-Ma** [comment]
Hi @anais2390 ,

Thanks for your feedback.
I think that because we use different reduction computations for Nan in metrics and loss:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/losses/dice.py#L182
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/utils.py#L74
And it should be expected as loss and metrics have different meanings for Nan, we ignore nans in metrics as some labels may be missing.

Thanks.
